be/src/storage/lake/compaction_scheduler.cpp

```diff
@@ -197,6 +206,9 @@ void CompactionScheduler::compact(::google::protobuf::RpcController* controller,
     }
     // initialize last check time, compact request is received right after FE sends it, so consider it valid now
     cb->set_last_check_time(time(nullptr));
+    // FIXME: be noticed that even the `_stopped` is checked in the beginning, it is still possible that the scheduler
+    // turns to stop status after the check and before this point. In case of the conflict happens, the contexts_vec
+    // may never be processed which leads to rpc hanging without response.
```

Comment:
should be done by abort all?

Reply:
so comment can be adjusted

Reply:
np. there is a gap between the `abort_all` and `compact()`. Consider thread A and B done in the following seq

```
1. thread-A: compact() execute to line 183
2. thread-B: stop()
3. thread-A: run to line 212
```
So the `abort_all()` is done and new context are still pushed into the task queue after abort_all.

Reply:
i think it can be handled? by adding a mutex in compaction scheduler?

Reply:
done